compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -577,11 +577,16 @@ object desugar {
           else
             enumApplyResult(cdef, parents, derivedEnumParams, appliedRef(enumClassRef, derivedEnumParams))
 
+        def applyShouldInheritAccess = {
```

Comment:
Not a fan of this method name, maybe just call it `restrictedAccess` with a documentation comment explaining what it implies exactly ?